src/NCalc/LambdaExpressionVistor.cs

```diff
@@ -26,6 +26,37 @@ internal class LambdaExpressionVistor : LogicalExpressionVisitor
             { typeof(ulong), new HashSet<Type> { typeof(float), typeof(double), typeof(decimal) }},
         };
 
+        private struct MathCallFunction
+        {
+            public string MethodName;
+            public int ArgumentCount;
+        }
+
+        private readonly Dictionary<string, MathCallFunction> _mathCallFunctions = new Dictionary<string, MathCallFunction>()

```

Comment:
@eugenca, In regards to only using reflection once, here's what I was thinking.  _mathCallFunctions can be static so that it is only instantiated and initialized once.  The initialization can also look up the method info, since you're only ever calling the overload that takes double(s). 

```
private struct MathCallFunction
{
    public MethodInfo MethodInfo;
    public int ArgumentCount;
}

private static readonly Dictionary<string, MathCallFunction> _mathCallFunctions = new() 
{
  { "ABS", new MatchCallFunction(typeof(Math).GetMethod(nameof(Math.Abs), new[typeof(double)]), 1) },
  //...
  { "ATAN2", new MatchCallFunction(typeof(Math).GetMethod(nameof(Math.Atan2), new[typeof(double), typeof(double)]), 2) },
  //...
};
```

Reply:
Looks reasonable